crates/bevy_ui/src/render/mod.rs

```diff
@@ -984,6 +987,154 @@ pub fn extract_text_shadows(
     }
 }
 
+fn extract_text_outlines(
+    mut aa_glyph_cache: Local<Vec<ExtractedGlyph>>,
+    mut commands: Commands,
+    mut extracted_uinodes: ResMut<ExtractedUiNodes>,
+    texture_atlases: Extract<Res<Assets<TextureAtlasLayout>>>,
+    uinode_query: Extract<
+        Query<(
+            Entity,
+            &ComputedNode,
+            &ComputedNodeTarget,
+            &UiGlobalTransform,
+            &InheritedVisibility,
+            Option<&CalculatedClip>,
+            &TextLayoutInfo,
+            &TextOutline,
+        )>,
+    >,
+    camera_map: Extract<UiCameraMap>,
+) {
+    aa_glyph_cache.clear();
+    let mut start = extracted_uinodes.glyphs.len();
+    let mut len = 0;
+
+    let mut camera_mapper = camera_map.get_mapper();
+    for (
+        entity,
+        uinode,
+        target,
+        global_transform,
+        inherited_visibility,
+        clip,
+        text_layout_info,
+        outline,
+    ) in &uinode_query
+    {
+        // Skip if not visible or if size is set to zero (e.g. when a parent is set to `Display::None`)
+        if !inherited_visibility.get() || uinode.is_empty() || outline.width.as_i32() == 0 {
+            continue;
+        }
+
+        let Some(extracted_camera_entity) = camera_mapper.map(target) else {
+            continue;
+        };
+
+        let width = (outline.width.as_f32() / uinode.inverse_scale_factor()).ceil() as i32;
+        let width_pow2 = width.pow(2);
+        let aa_factor = outline.anti_aliasing.unwrap_or(1.0);
+        let color: LinearRgba = outline.color.into();
+        let mut aa_color = color;
+        aa_color.alpha *= aa_factor;
+
+        for (
+            i,
+            PositionedGlyph {
+                position,
+                atlas_info,
+                ..
+            },
+        ) in text_layout_info.glyphs.iter().enumerate()
+        {
+            let rect = texture_atlases
+                .get(&atlas_info.texture_atlas)
+                .unwrap()
+                .textures[atlas_info.location.glyph_index]
+                .as_rect();
+
+            for offset_x in -width..=width {
+                // Adjust height to follow a radial pattern.
+                let height = ((width_pow2 - offset_x.pow(2)).abs() as f32).sqrt().ceil() as i32;
+
+                for offset_y in -height..=height {
+                    if offset_x == 0 && offset_y == 0 {
+                        continue;
+                    }
+
+                    let offset = Vec2 {
+                        x: offset_x as f32,
+                        y: offset_y as f32,
+                    };
+
+                    let transform = Affine2::from(*global_transform)
+                        * Affine2::from_translation(
+                            -0.5 * uinode.size() + offset / uinode.inverse_scale_factor(),
```

Comment:
```suggestion
                            -0.5 * uinode.size() + offset,
```
It's been moved to the width now.